slave.mk

```diff
@@ -539,7 +539,7 @@ $(SONIC_INSTALL_DEBS) : $(DEBS_PATH)/%-install : .platform $$(addsuffix -install
 			{ while dpkg -s $(firstword $(subst _, ,$(basename $(deb)))) | grep "^Version: $(word 2, $(subst _, ,$(basename $(deb))))" &> /dev/null; do echo "waiting for $(deb) to be uninstalled" $(LOG); sleep 1; done } )
 		# put a lock here because dpkg does not allow installing packages in parallel
 		if mkdir $(DEBS_PATH)/dpkg_lock &> /dev/null; then
-			{ sudo DEBIAN_FRONTEND=noninteractive dpkg -i $(DEBS_PATH)/$* $(LOG) && rm -d $(DEBS_PATH)/dpkg_lock && break; } || { rm -d $(DEBS_PATH)/dpkg_lock && exit 1 ; }
+			{ sudo DEBIAN_FRONTEND=noninteractive dpkg -i $(DEBS_PATH)/$* $(LOG) && rm -d $(DEBS_PATH)/dpkg_lock && break; } || { rm -d $(DEBS_PATH)/dpkg_lock && sleep 1 ; }
```

Comment:
can you explain why you think the root cause is, and why this would fix the problem?

Reply:
https://dev.azure.com/mssonic/build/_build?definitionId=154&_a=summary
In this pipeline, I set timeout=60, and triggerred 8 build at Apr. 22. Build 1-4 runs without this change, and 3 of 4 break at target "libhiredis". Build 5-8 runs with this change, and all of them passed target "libhiredis".

Reply:
When 'dpkg -i *' exit with non-zero code. We don't need to exit. Just run it agin, it will succeed. Because the most probably reason is parallel dpkg installation.

Reply:
I have the same question and could not understand.

Reply:
I didn't find somewhere else which hold dpkg frontend lock.
But error is reported at this line. 
[AZP](https://dev.azure.com/mssonic/build/_build?definitionId=182&_a=summary) shows this change works.

Reply:
[ FAIL LOG START ] [ target/debs/buster/libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb-install ]
dpkg: error: dpkg frontend lock is locked by another process
[  FAIL LOG END  ] [ target/debs/buster/libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb-install ]
[ finished ] [ target/debs/buster/libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb-install ] 
[ FAIL LOG START ] [ target/debs/buster/libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb-install ]
dpkg: error: dpkg frontend lock is locked by another process
dpkg: error: dpkg frontend lock is locked by another process
[  FAIL LOG END  ] [ target/debs/buster/libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb-install ]
[ finished ] [ target/debs/buster/libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb-install ] 
[ FAIL LOG START ] [ target/debs/buster/libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb-install ]
dpkg: error: dpkg frontend lock is locked by another process
dpkg: error: dpkg frontend lock is locked by another process
dpkg: error: dpkg frontend lock is locked by another process
[  FAIL LOG END  ] [ target/debs/buster/libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb-install ]
[ finished ] [ target/debs/buster/libhiredis0.14_0.14.0-3~bpo9+1_amd64.deb-install ]